presto-base-jdbc/src/test/java/com/facebook/presto/plugin/jdbc/optimization/TestJdbcJoinPushdown.java

```diff
@@ -0,0 +1,280 @@
+import com.facebook.presto.spi.plan.PlanNode;
+import com.facebook.presto.spi.plan.PlanNodeIdAllocator;
+import com.facebook.presto.spi.plan.TableScanNode;
+import com.facebook.presto.spi.relation.VariableReferenceExpression;
+import com.facebook.presto.sql.planner.Plan;
+import com.facebook.presto.sql.planner.TypeProvider;
+import com.facebook.presto.sql.planner.assertions.MatchResult;
+import com.facebook.presto.sql.planner.assertions.Matcher;
+import com.facebook.presto.sql.planner.assertions.PlanAssert;
+import com.facebook.presto.sql.planner.assertions.PlanMatchPattern;
+import com.facebook.presto.sql.planner.assertions.SymbolAliases;
+import com.facebook.presto.sql.planner.iterative.rule.test.PlanBuilder;
+import com.facebook.presto.sql.tree.SymbolReference;
+import com.facebook.presto.testing.TestingConnectorSession;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import org.testng.annotations.Test;
+
+import java.sql.Types;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static com.facebook.presto.SessionTestUtils.TEST_SESSION;
+import static com.facebook.presto.common.type.BigintType.BIGINT;
+import static com.facebook.presto.common.type.BooleanType.BOOLEAN;
+import static com.facebook.presto.metadata.MetadataManager.createTestMetadataManager;
+import static com.facebook.presto.sql.planner.assertions.PlanMatchPattern.node;
+import static com.google.common.base.Preconditions.checkState;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static java.util.function.Function.identity;
+import static java.util.stream.Collectors.toMap;
+
+public class TestJdbcJoinPushdown
+{
+    private final JdbcJoinPushdown jdbcJoinPushdown;
+
+    private static final Metadata METADATA = createTestMetadataManager();
+    private static final PlanBuilder PLAN_BUILDER = new PlanBuilder(TEST_SESSION, new PlanNodeIdAllocator(), METADATA);
+    private static final String CATALOG_NAME = "Jdbc";
+    private static final String CONNECTOR_ID = new ConnectorId(CATALOG_NAME).toString();
+    private static final PlanNodeIdAllocator ID_ALLOCATOR = new PlanNodeIdAllocator();
+
+    public TestJdbcJoinPushdown()
+    {
+        this.jdbcJoinPushdown = new JdbcJoinPushdown();
+    }
+
+    @Test
+    public void testJdbcJoinPushdownWithJoins()
+    {
+        PlanNode original = combinedJdbcTableScan(BIGINT, "c1", "c2");
+        ConnectorSession session = new TestingConnectorSession(ImmutableList.of());
+        PlanNode actual = this.jdbcJoinPushdown.optimize(original, session, null, ID_ALLOCATOR);
+
+        final String aliasPrefix = "T";
+        List<ConnectorTableHandle> tableHandles = new ArrayList<>();
```

Comment:
Let's rename this to be more descriptive `expectedJoinTableHandles`, please use a ImmutableList builder for this